modules/core/src/main/java/dev/acog/bukkit/language/core/BukkitLanguage.java

```diff
@@ -1,66 +1,79 @@
 package dev.acog.bukkit.language.core;
 
+import dev.acog.bukkit.language.core.utils.FileResourcesUtils;
 import org.bukkit.entity.Player;
+import org.bukkit.plugin.java.JavaPlugin;
 
 import java.io.File;
 import java.io.FilenameFilter;
+import java.nio.file.Path;
+import java.util.List;
 import java.util.NoSuchElementException;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class BukkitLanguage {
 
-    private final File path;
+    private final File file;
     private final String defaultLocate;
     private final ConcurrentHashMap<String, LocateLanguage> languages = new ConcurrentHashMap<>();
 
-    private BukkitLanguage(File path, String locate) {
-        this.path = path;
+    private BukkitLanguage(File file, String locate) {
+        this.file = file;
         this.defaultLocate = locate.toLowerCase();
     }
 
+    /**
+     * This method creates and returns a BukkitLanguage instance by loading the language
+     * file located at the specified path and using the given locate parameter.
+     * @param plugin JavaPlugin Class
+     * @param locale ISO-3166 + ISO-3166
+     * @param file language pack path
+     * @return BukkitLanguage.class
+     */
+    public static BukkitLanguage load(JavaPlugin plugin, String locale, File file) {
+        List<Path> resources = FileResourcesUtils.getPathsFromResourceJar(file);
+        if (resources != null) {
+            for (Path path : resources) {
+                plugin.saveResource(path.toString(), false);
+            }
+        }
+        return new BukkitLanguage(file, locale).reload();
+    }
+
     public BukkitLanguage reload() {
         FilenameFilter filter = (dir, name) -> name.endsWith(".yml");
-        File[] files = this.path.listFiles(filter);
+        File[] files = file.listFiles(filter);
+
         if (files == null || files.length == 0) {
             throw new NoSuchElementException("No language files found in the directory.");
         }
-        for (File locatePath : files) {
-            String locate = locatePath.getName().replace(".yml", "").toLowerCase();
-            languages.put(locate, LocateLanguage.load(locate, locatePath));
+        for (File locateFile : files) {
+            String locate = locateFile.getName().replace(".yml", "").toLowerCase();
```

Comment:
p2: .yml.yml 같은 파일이 있으면 잘못 동작할 것 같아요. 
```suggestion
            String locate = locateFile.getName().substring(0, locateFile.length() - 4).toLowerCase();
```

Reply:
피드백을 받아 적절하게 반영 했습니다.